src/outputtype.h

```diff
@@ -28,7 +29,7 @@ static constexpr auto OUTPUT_TYPES = std::array{
     OutputType::BECH32M,
 };
 
-[[nodiscard]] bool ParseOutputType(const std::string& str, OutputType& output_type);
+std::optional<OutputType> ParseOutputType(const std::string& str);
```

Comment:
Is there a reason for removing the `[[nodiscard]]` attribute?

Reply:
I'm not sure why we'd keep it now that we're returning the result. If it's unused the compiler should emit `Wunused-result `?

Reply:
Makes sense